Examples/Algorithms/Generators/ActsExamples/Generators/EventGenerator.hpp

```diff
@@ -44,13 +44,30 @@ class EventGenerator final : public ActsExamples::IReader {
   /// The process generator is responsible for defining all components of the
   /// particle barcode except the primary vertex. The primary vertex will be
   /// set/overwritten by the event generator.
-  using MultiplicityGenerator = std::function<size_t(RandomEngine&)>;
-  using VertexGenerator = std::function<Acts::Vector4(RandomEngine&)>;
-  using ParticlesGenerator = std::function<SimParticleContainer(RandomEngine&)>;
+  // using MultiplicityGenerator = std::function<size_t(RandomEngine&)>;
+  // using VertexGenerator = std::function<Acts::Vector4(RandomEngine&)>;
+  // using ParticlesGenerator =
+  // std::function<SimParticleContainer(RandomEngine&)>;
+
+  struct MultiplicityGenerator {
+    virtual ~MultiplicityGenerator() = default;
+    virtual size_t operator()(RandomEngine& rng) const = 0;
+  };
+
+  struct VertexGenerator {
+    virtual ~VertexGenerator() = default;
+    virtual Acts::Vector4 operator()(RandomEngine& rng) const = 0;
+  };
+
+  struct ParticlesGenerator {
+    virtual ~ParticlesGenerator() = default;
+    virtual SimParticleContainer operator()(RandomEngine& rng) = 0;
```

Comment:
Why is it not `const`? Maybe add some description here?

Reply:
You're right, should be const. I also added some doc comments.

Reply:
On second thought, the Pythia8 generator's `operator()` can't be const because it uses Pythia8. I added this to the explanation here.

Reply:
I see. Thanks!